ibis/backends/base/sql/alchemy/client.py

```diff
@@ -329,3 +330,117 @@ def _sqla_table_to_expr(self, table):
+
+        Raises
+        -------
+        NotImplementedError
+            Inserting data to a table from a different database is not
+            yet implemented
+
+        ValueError
+            You must pass obj variable (pandas Dataframe or table name)
+            to insert data from
+
+        TypeError
+            No operation is being performed. Either the obj parameter
+            is not a pandas dataframe or is not of string datatype.
+
+        """
+
+        if database == self.database_name:
+            # avoid fully qualified name
+            database = None
+
+        if database is not None:
+            raise NotImplementedError(
+                'Inserting data to a table from a different database is not '
+                'yet implemented'
+            )
+
+        if obj is None:
+            raise ValueError(
+                'You must pass obj variable (pandas Dataframe or table name)'
+                ' to insert data from'
+            )
+
+        params = {}
+        if self.has_attachment:
+            # for database with attachment
+            # see: https://github.com/ibis-project/ibis/issues/1930
+            params['schema'] = self.database_name
+
+        if isinstance(obj, pd.DataFrame):
+            if overwrite:
+                obj.to_sql(
+                    table_name,
+                    self.con,
+                    index=False,
+                    if_exists='replace',
+                    **params,
+                )
+            else:
+                obj.to_sql(
+                    table_name,
+                    self.con,
+                    index=False,
+                    if_exists='append',
+                    **params,
+                )
+        elif isinstance(obj, str) or isinstance(obj, ir.TableExpr):
+            to_table_expr = self.table(table_name)
+            to_table_schema = to_table_expr.schema()
+
+            if overwrite:
+                self.drop_table(table_name, database=database)
+                self.create_table(
+                    table_name, schema=to_table_schema, database=database,
+                )
+
+            to_table = self._get_sqla_table(table_name, schema=database)
+
+            from_table_expr = None
+            if isinstance(obj, str):
+                from_table_expr = self.table(obj)
+            elif isinstance(obj, ir.TableExpr):
+                from_table_expr = obj
+
+            with self.begin() as bind:
+                if from_table_expr is not None:
+                    bind.execute(
+                        to_table.insert().from_select(
+                            list(from_table_expr.columns),
+                            from_table_expr.compile(),
+                        )
+                    )
+        else:
+            raise TypeError(
+                'No operation is being performed. Either the obj'
+                ' parameter is not a pandas dataframe or'
+                ' is not of string datatype.'
```

Comment:
Or an Ibis `TableExr` now, right?

Reply:
Yes. My bad missed it in the message. Updated!